From d68fe13ce880e60f41a00718724cbd3edbeeb0f5 Mon Sep 17 00:00:00 2001
From: Aleksander Jan Bajkowski <olek2@wp.pl>
Date: Sat, 5 Jul 2025 15:14:50 +0200
Subject: [PATCH] net: phy: realtek: add RTL8261N PHY

Signed-off-by: Aleksander Jan Bajkowski <olek2@wp.pl>
---
 drivers/net/phy/realtek/realtek_main.c | 48 ++++++++++++++++++++++++++
 1 file changed, 48 insertions(+)

diff --git a/drivers/net/phy/realtek/realtek_main.c b/drivers/net/phy/realtek/realtek_main.c
index dd0d675149ad..c91dd4b14a3b 100644
--- a/drivers/net/phy/realtek/realtek_main.c
+++ b/drivers/net/phy/realtek/realtek_main.c
@@ -159,6 +159,7 @@
 #define RTL_8221B_VN_CG				0x001cc84a
 #define RTL_8251B				0x001cc862
 #define RTL_8261C				0x001cc890
+#define RTL_8261N				0x001ccaf3
 
 /* RTL8211E and RTL8211F support up to three LEDs */
 #define RTL8211x_LED_COUNT			3
@@ -1095,6 +1096,17 @@ static int rtl822xb_config_init(struct phy_device *phydev)
 	return phy_write_mmd(phydev, MDIO_MMD_VEND1, 0x6f11, 0x8020);
 }
 
+static int rtl8261b_config_init(struct phy_device *phydev)
+{
+	int ret;
+
+	ret = phy_write_mmd(phydev, MDIO_MMD_VEND1, 0x0141, 0x8003);
+	if (ret < 0)
+		return ret;
+
+	return (phy_write_mmd(phydev, MDIO_MMD_VEND1, 0x0143, 0x8C07));
+}
+
 static int rtl822xb_get_rate_matching(struct phy_device *phydev,
 				      phy_interface_t iface)
 {
@@ -1401,6 +1413,18 @@ static int rtl8251b_c45_match_phy_device(struct phy_device *phydev,
 	return rtlgen_is_c45_match(phydev, RTL_8251B, true);
 }
 
+static int rtl8261n_c22_match_phy_device(struct phy_device *phydev,
+					 const struct phy_driver *phydrv)
+{
+	return rtlgen_is_c45_match(phydev, RTL_8261N, false);
+}
+
+static int rtl8261n_c45_match_phy_device(struct phy_device *phydev,
+					 const struct phy_driver *phydrv)
+{
+	return rtlgen_is_c45_match(phydev, RTL_8261N, true);
+}
+
 static int rtlgen_resume(struct phy_device *phydev)
 {
 	int ret = genphy_resume(phydev);
@@ -1753,6 +1777,30 @@ static struct phy_driver realtek_drvs[] = {
 		.resume         = rtlgen_resume,
 		.read_page      = rtl821x_read_page,
 		.write_page     = rtl821x_write_page,
+	}, {
+		.match_phy_device = rtl8261n_c22_match_phy_device,
+		.name           = "RTL8261N 10Gbps PHY (c22)",
+		.probe		= rtl822x_probe,
+		.get_features   = rtl822x_get_features,
+		.config_aneg    = rtl822x_config_aneg,
+		.config_init    = rtl822xb_config_init,
+		.get_rate_matching = rtl822xb_get_rate_matching,
+		.read_status    = rtl822xb_read_status,
+		.suspend        = genphy_suspend,
+		.resume         = rtlgen_resume,
+		.read_page      = rtl821x_read_page,
+		.write_page     = rtl821x_write_page,
+	}, {
+		.match_phy_device = rtl8261n_c45_match_phy_device,
+		.name           = "RTL8261N 10Gbps PHY (c45)",
+		.probe		= rtl822x_probe,
+		.config_init    = rtl8261b_config_init,
+		.get_rate_matching = rtl822xb_get_rate_matching,
+		.get_features   = rtl822x_c45_get_features,
+		.config_aneg    = rtl822x_c45_config_aneg,
+		.read_status    = rtl822xb_c45_read_status,
+		.suspend        = genphy_c45_pma_suspend,
+		.resume         = rtlgen_c45_resume,
 	}, {
 		.match_phy_device = rtl_internal_nbaset_match_phy_device,
 		.name           = "Realtek Internal NBASE-T PHY",
-- 
2.47.3

