From ad9c4a962339fad5afc998f4cf3546cc49caaf6f Mon Sep 17 00:00:00 2001
From: Aleksander Jan Bajkowski <olek2@wp.pl>
Date: Sun, 6 Jul 2025 00:07:55 +0200
Subject: [PATCH] test

---
 drivers/net/phy/sfp.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/phy/sfp.c b/drivers/net/phy/sfp.c
index 707f65d65694..b5c0ddb95f73 100644
--- a/drivers/net/phy/sfp.c
+++ b/drivers/net/phy/sfp.c
@@ -425,6 +425,8 @@ static void sfp_fixup_rollball_cc(struct sfp *sfp)
 static void sfp_fixup_cc(struct sfp *sfp)
 {
 
+	sfp->module_t_start_up = msecs_to_jiffies(60000);
+	sfp->module_t_wait = msecs_to_jiffies(60000);
 	/* Some RollBall SFPs may have wrong (zero) extended compliance code
 	 * burned in EEPROM. For PHY probing we need the correct one.
 	 */
-- 
2.39.5

